cmd/vulcan-exposed-http/main.go

```diff
@@ -79,56 +80,50 @@ var (
 		"18091", // memcached Internal REST HTTPS for SSL
 		"18092", // memcached Internal CAPI HTTPS for SSL
 	}
-
-	exposedVuln = report.Vulnerability{
-		Summary:     "Exposed HTTP Port",
-		Description: "An HTTP server is listening at least in one port ot the server.",
-		Score:       report.SeverityThresholdNone,
-	}
 )
 
-func exposedHTTP(target string, nmapReport *gonmap.NmapRun) []report.Vulnerability {
-	gr := report.ResourcesGroup{
-		Name: "Network Resources",
-		Header: []string{
-			"Hostname",
-			"Port",
-			"Protocol",
-			"Service",
-			"Version",
-			"SSL",
-		},
-	}
-
-	add := false
+func exposedHTTP(target string, nmapReport *gonmap.NmapRun, state checkstate.State) {
 	for _, host := range nmapReport.Hosts {
 		for _, port := range host.Ports {
 			if port.State.State != "open" || !strings.Contains(port.Service.Name, "http") {
 				continue
 			}
-
-			add = true
-
-			networkResource := map[string]string{
-				"Hostname": target,
-				"Port":     strconv.Itoa(port.PortId),
-				"Protocol": port.Protocol,
-				"Service":  port.Service.Product,
-				"Version":  port.Service.Version,
-			}
-			if port.Service.Tunnel == "ssl" {
-				networkResource["SSL"] = "yes"
-			}
-			gr.Rows = append(gr.Rows, networkResource)
+			v := report.Vulnerability{
+				AffectedResource: fmt.Sprintf("%s:%d", target, port.PortId),
+				Labels:           []string{port.Protocol, "issue"},
+				Fingerprint:      helpers.ComputeFingerprint(port),
+				Summary:          "Exposed HTTP Port",
+				Description:      "An HTTP server is listening at least in one port ot the server.",
+				Score:            report.SeverityThresholdNone,
+				Resources: []report.ResourcesGroup{{
```

Comment:
This is totally a matter of taste, but I'd separate the `{` on different lines, like:
```
Resources: []report.ResourcesGroup{
    {
    ...
    },
}
```

Reply:
Sure